src/main/java/com/autotune/analyzer/services/CreateExperiment.java

```diff
@@ -98,6 +100,26 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
                 for (CreateExperimentAPIObject createExperimentAPIObject : createExperimentAPIObjects) {
                     createExperimentAPIObject.setExperiment_id(Utils.generateID(createExperimentAPIObject.toString()));
                     createExperimentAPIObject.setStatus(AnalyzerConstants.ExperimentStatus.IN_PROGRESS);
+                    // updating experiment type to container if not passed
```

Comment:
Avoid Redundant Calls: Instead of calling createExperimentAPIObject.getExperimentType() multiple times, store it in a local variable.
Boolean Flags for Type Checks: Using boolean flags like isContainerExperiment and isNamespaceExperiment improves code readability.
String Comparison: Used AnalyzerConstants.REMOTE.equalsIgnoreCase() for null-safe comparison.
Null Check Simplification: Combined null checks to reduce redundancy and improve performance.
This should make the code more efficient and easier to maintain.

Reply:
Thanks for suggesting the optimizations! I've updated the code as suggested.